src/lib/eventPool.js

```diff
@@ -0,0 +1,63 @@
+import _ from 'lodash'
+
+class EventPool {
+  events = []
+  queues = {}
+
+  // ------------------------------------
+  // Utils
+  // ------------------------------------
+
+  emit = name => event => {
+    _.forEach(this.queues, (queue, name) => {
+      const { [name]: handlers } = queue
+      if(handlers) _.forEach(handlers, handler => handler(event))
+    })
+  }
+
+  normalize = handlers => _.isArray(handlers) ? handlers : [handlers]
+
+  // ------------------------------------
+  // Listeners handling
+  // ------------------------------------
+
+  listen = name => {
+    if(_.includes(this.queues, name)) return
+    document.addEventListener(name, this.emit(name))
+  }
+
+  unlisten = name => {
+    if(_.some(this.queues, name)) return
+    document.removeEventListener(name, this.emit(name))
```

Comment:
@davezuko Yep, you're fully right there. I pushed some changes and tests, now it works as I described.